File: CODEJAM/16_RoundC/3.cpp
#include<iostream>
#include<math>
using namespace std;
#define ll long long int
int answer(int n,int k)
{
	double temp,temp1;
	ll tl = 1,tl1;
	temp1 = (double)k/100.0;
	while(1)
	{
		temp = math.sqrt(2*tl*n - tl*tl);
		tl1 = (int)temp;
		temp = n- math.sqrt((n-tl)*(n-tl)+tl1*tl1);
		if(temp>temp1)

	}
}
int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		ll n,k;
		cin>>n>>k;
		double temp;
		double temp1;
		ll tl = 1,tl1;
		temp1 = (double)k/100.0;
		int check = 0;
		while(tl<n/2)
		{
			temp = math.sqrt(2*tl*n - tl*tl);
			tl1 = (int)temp;
			temp = n- math.sqrt((n-tl)*(n-tl)+tl1*tl1);
			if(temp>temp1)
			{
				check=1;
				break;
			}	
			tl++;	
		}
		if(check)
			cout<<n-tl<<" "<<tl1;
		else
			cout<<"-1"<<endl;
	}	
}

File: CODEJAM/16_RoundC/3.py
from collections import defaultdict

def check(dict1):
	for each in dict1:


def main():
	t = int(input)
	i = 0
	while i<t:
		i++
		n = int(input)
		string = []
		variables = []
		dict1 = defaultdict(lambda:list)
		for i in range(n):
			string.append(raw_input())
			temp1 = string[i].split("=")
			temp = temp1[1]
			j = temp.find("(")
			k = temp.find(")")
			temp = temp[j+1:k]
			dict1[temp1[0]].append(temp.split(",")) 

		
		ans = check(dict1)